0032-peering-clearing-settlement/0032-peering-clearing-settlement.md

```diff
@@ -0,0 +1,160 @@
+> - After Step 6:
+>   - Alice's balance is 14 satoshis
+>   - Bob's balance is -14 satoshis
+
+# Clearing
+
+The traditional definition of clearing a payment, is the process by which the two parties establish their net positions and reconcile any differences.
+
+In the Interledger case this would involve the two peers exchanging data on what their current balance is and, if there is a discrepency, resolving this.
+
+Notably, clearing is not part of the Interledger standard, as the means by which two peers perform these operations is a bilateral concern (i.e. has no impact on others in the network).
+
+Current implementations of ILP do not have a standardized clearing process, as most peers settle very frequently using networks on which this is possible such as distributed ledgers which support payment channels, reducing the risk of discrepencies developing in their account balances.
+
+However, there is an edge case where the peer's balances may be out of sync demonstrated in this example.
+
+> **Example : Alice and Bob**
+
+> Assuming they start with a balance of 0 and the following sequence of events occurs:
+
+> 1. Bob passes an ILP Prepare to Alice with an amount of 20.
+> 1. Alice replies with a valid ILP Fulfill and believes she has done so before the request expires, however Bob considers the request to have expired and has already rejected the payment to his downstream peer.
+
+> The balance changes will be as follows:
+
+> - After Step 2:
+>   - Alice's balance is 20 satoshis
+>   - Bob's balance is 0 satoshis
+
+> Clearly, Bob and Alice need to reconcile this situation. The specific mechanism they use and recourse they take to resolve this difference is outside the scope of the Interledger protocol. 
+
+It is important to note that this is an extremely unlikely scenario as telescoping expiry timeouts between peers should reduce the risk of this to almost zero.
+
+Further, this is a single packet, and a key principal of the Interledger network is to break payments into very small packets so that in the unlikely event that this does occur the losses incurred by the two parties are negligible.
+
+Nonetheless, it is likely that future bilateral protocols (like BTP) will define more detailed processes for reconciliation and clearing.
+
+# Settlement
+
+It is unlikely that the flow of packets between two peers will always net out such that the balance continuously tends toward zero. In most cases flows will be predominantly in one direction and there will be a need for the peers to reduce them back to 0.
+
+This avoids a situation where one peer owes the other a large sum of money and is then unable or unwilling to pay (settlement risk). Connectors should define a maximum Interledger account balance to manage the settlement risk they expose themselves to with their peers.
+
+Further, the principal of the Interledger network is to settle as often as is practically possible and financially viable such that this risk is very low.
```

Comment:
I'm not sure that's a principle (also typo with that word) of the network, but more like a mechanism we've found very helpful especially between peers that do not have a lot of bilateral credit/trust. Over time that could change